praw/models/reddit/wikipage.py

```diff
@@ -49,6 +49,29 @@ def add(self, redditor: "praw.models.Redditor"):
         )
         self.wikipage._reddit.post(url, data=data)
 
+    def hide_revision(self) -> Dict[str, bool]:
+        """Toggle the public visibility of a wikipage revision.
+
+        To hide or show revision ``[ID]`` of ``"praw_test"`` in ``r/test``, try
+
+        .. code-block:: python
+
+            reddit.subreddit("test").wiki["praw_test"].revision("[ID]").mod.hide_revision()
+
+        This will return ``{'status': True}`` if the revision has been hidden, and
+        ``{'status': False}`` if the revision has been unhidden.
+
+        A hidden revision will still be visible to other moderators.
+
+        """
+        return self.wikipage._reddit.post(
```

Comment:
Could we check if a revision is not hidden first (using the `revision_hidden` attribute) and only send a request when it isn't already hidden? Then add a method that unhides the revision. My reasoning is because `hide_revision` implies that the revision in question will be hidden and to me it's counter intuitive call this method on a revision that is hidden and have it become unhidden.

Reply:
How about adding a parameter to `hide` so that the user can specify to only hide or only unhide? The method would then check to the status of the revision before sending out a request. Or perhaps `hide` could be split out into `hide` and `unhide`? What should happen if the check were false?

The trouble with doing this is that, as far as I can tell, WikiPages don't have a `revision_hidden` attribute. That attribute is only present in the listing of revisions. If you fetch a page by its revision id, the only `revision_` attributes given are `revision_date`, `revision_by`, and `revision_id`. The fact that `revision_hidden` is missing looks like an oversight by Reddit.    

For the time being, I could tweak the definition of WikiPage so that it has a revision_hidden property, and then have [_revision_generator](https://github.com/praw-dev/praw/blob/master/praw/models/reddit/wikipage.py#L137) set this property. If someone tries to hide a revision that they've accessed directly, we could determine the revision's visibility by sending one request to /wiki/revisions/<PAGE>?before=WikiRevision_<ID>, and then a second request to /wiki/revisions/<PAGE>?after=WikiRevision_<before_ID>. If the page doesn't have either `before` or `after` attributes, a call would just be made to /wiki/revisions/<PAGE>.

Reply:
Ah I see, I didn't know that it only returned `revision_hidden` from the revisions list. 

> Or perhaps hide could be split out into hide and unhide? 

I like this idea.

> What should happen if the check were false?

We could have `hide` and `unhide` issue a second request if it returns `False` or `True` respectively so that the desired state can be easily achieved. With that, we would need a note in there indicating that behavior and the potential for a second request. 

The other thing we could have is have a singular method like `toggle_visibility` that can do both with a `hide` parameter that will: ensure it is hidden if `True`, ensure it is visible if `False`, or just toggle if it isn't provided or if it is set to `None`.